docs/dev/checks/README.md

```diff
@@ -133,3 +125,91 @@ checks code.
+
+The list of checks currently shipping with the agent lives in the
+[integrations-core repo
+here](https://github.com/DataDog/integrations-core/blob/master/requirements-agent-release.txt)
+
+Say you want to test the `openmetrics` check.
+
+1. Clone `integrations-core` locally. (Optionally checkout the git tag
+   corresponding to the version you want to test)
+2. From integration-core directory, install `datadog-checks-base` as a pre-req.
+   `pip3.10 install --user './datadog_checks_base[deps]'`
+3. Install the `openmetrics` check. From inside the `integrations-core`
+   checkout:
+   `pip3.10 install --user ./openmetrics`
+
+That's it! Your local build should now have the correct packages to be able to
+run the `openmetrics` check.
+
+#### "What is this `[deps]` thing?"
+The `[deps]` at the end of the package name instructs pip to install the
+"requires" that match the `deps` "extra". Without this, you'll see errors when running the agent
+about missing dependencies.
+
+View all the possible 'extras' with:
+```
+pip3.10 install --no-warn-script-location --user --dry-run --ignore-installed -qqq --report - datadog-checks-base | jq '.install[] | select(.metadata.name=="datadog-checks-base") | .metadata.provides_extra'
+```
+
+View all the "requires" (which shows which "extra" will trigger that
+dependency to be installed) with:
+```
+pip3.10 install --no-warn-script-location --user --dry-run --ignore-installed -qqq --report - datadog-checks-base | jq '.install[] | select(.metadata.name=="datadog-checks-base") | .metadata.requires_dist'
+```
+
+#### "Which python is my agent build using?"
+Assuming you're not doing a full omnibus build, you won't have an embedded
+python in your local build, so your local build will need to choose a python
+from your system to use.
+
+By default the agent will choose the first python binary (`python` or `python3`
+depending on your configuration) from the `PATH` that is used to run the agent.
+
+### "Could not initialize Python"
+Out of the box, after an `inv agent.build`, you may see the following error on
+linux when trying to run the resulting `agent` binary:
+
+`Could not initialize Python: could not load runtime python for version 3: Unable to open three library: libdatadog-agent-three.so: cannot open shared object file: No such file or directory`
+
+To solve on linux, use the loader env var `LD_LIBRARY_PATH`.
+ie, `LD_LIBRARY_PATH=$PWD/dev/lib ./bin/agent/agent run`
+
+Why is this needed? This is due to a combination of the way `rtloader` works
+and how library loading works on linux.
+
+The very simplified summary is that `libdatadog-agent-rtloader.so` attempts to load
+`libdatadog-agent-three.so` via `dlopen`, however `libdatadog-agent-rtloader`
+does not have its `RUNPATH` set correctly in local builds, so `dlopen` is unable to find
+`libdatadog-agent-three.so`. Using `LD_LIBRARY_PATH` instructs `dlopen` where to
+search for libraries, so using it sidesteps this issue.
+
```

Comment:
We could add some info on how to use `virtualenv` in the future to not pollute system python but this is great stuff as-is!

Reply:
Yes I'm sure `virtualenv` would be cleaner here, but hopefully this serves as a jumping off point for some future person to write up how to use `virtualenv` correctly here